[Android] Dismissing a sheet may cause System.NullReferenceException

Observed on 8.0.2 version. This happen occasionally when dismissing a sheet programmatically or via gesture. Mainly after many open/close of the same sheet.

Observed on Android 10 (physical device).

For reference here's the underlaying exception:

```
[mono-rt] [ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object.
[mono-rt]    at The49.Maui.BottomSheet.ContentDetent.GetHeight(BottomSheet page, Double maxSheetHeight)
[mono-rt]    at The49.Maui.BottomSheet.BottomSheetController.CalculateHeights(Double maxSheetHeight)
[mono-rt]    at The49.Maui.BottomSheet.BottomSheetController.<OnLayoutChange>b__56_0()
[mono-rt]    at Microsoft.Maui.Dispatching.Dispatcher.<>c__DisplayClass4_0.<DispatchImplementation>b__0() in D:\a\_work\1\s\src\Core\src\Dispatching\Dispatcher.Android.cs:line 24
[mono-rt]    at Java.Lang.Thread.RunnableImplementor.Run() in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Java.Lang/Thread.cs:line 36
[mono-rt]    at Java.Lang.IRunnableInvoker.n_Run(IntPtr jnienv, IntPtr native__this) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Java.Lang.IRunnable.cs:line 84
[mono-rt]    at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PP_V(_JniMarshal_PP_V callback, IntPtr jnienv, IntPtr klazz) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:line 26
```

--- a/src/Models/ContentDetent.cs
+++ b/src/Models/ContentDetent.cs
@@ -4,15 +4,19 @@ public partial class ContentDetent : Detent
 {
     public override double GetHeight(BottomSheet page, double maxSheetHeight)
     {
-        if (page.Content is null)
+        if (page?.Content is null)
         {
             return maxSheetHeight;
         }
-        if (page.Content is ScrollView sv)
-        {
-            var s = sv.Content.Measure(page.Window.Width - page.Padding.HorizontalThickness, maxSheetHeight);
-        }
-        var r = page.Content.Measure(page.Window.Width - page.Padding.HorizontalThickness, maxSheetHeight);
+
+        // I left this code as comment because it might be useful for future reference, but it's not used
+        //if (page.Content is ScrollView sv)
+        //{
+        //    var s = sv.Content.Measure(page.Window.Width - page.Padding.HorizontalThickness, maxSheetHeight);
+        //}
+
+        // HACK: from 'page.Window.Width' to 'page.Window?.Width ?? page.Width' 
+        var r = page.Content.Measure(page.Window?.Width ?? page.Width - page.Padding.HorizontalThickness, maxSheetHeight);
 
         return Math.Min(maxSheetHeight, r.Request.Height + page.Padding.VerticalThickness);
     }

--- a/src/Platforms/Android/BottomSheetHandler.cs
+++ b/src/Platforms/Android/BottomSheetHandler.cs
@@ -13,7 +13,7 @@ public partial class BottomSheetHandler
 
     partial void Dismiss(BottomSheet view, object request)
     {
-        view.Controller?.Dismiss((bool)request);
+        view?.Controller?.Dismiss((bool)request);
     }
 
     partial void PlatformUpdateSelectedDetent(BottomSheet view)

--- a/src/Platforms/iOS/BottomSheetHandler.cs
+++ b/src/Platforms/iOS/BottomSheetHandler.cs
@@ -13,8 +13,8 @@ public partial class BottomSheetHandler
 
     partial void Dismiss(BottomSheet view, object request)
     {
-        view.CachedDetents.Clear();
-        view.Controller?.DismissViewController((bool)request, view.NotifyDismissed);
+        view?.CachedDetents.Clear();
+        view?.Controller?.DismissViewController((bool)request, view.NotifyDismissed);
     }
 
     partial void PlatformMapSelectedDetent(BottomSheet view)

